File: coresuite-dotnet-standard-vb/reportwriter/SimpleSubReport.vb
Imports ceTe.DynamicPDF
Imports ceTe.DynamicPDF.LayoutEngine
Imports coresuite_dotnet_reportdata_vb

Namespace coresuite_dotnet_standard_vb.reportwriter

    Public Class SimpleSubReport

        Public Shared Sub Run(outputPdfPath As String)
            ' Create the document's layout from a DLEX template
            Dim layoutReport As DocumentLayout = New DocumentLayout(Util.GetResourcePath("DLEX/SimpleSubReport.dlex"))

            Dim layoutData As NameValueLayoutData = New NameValueLayoutData()
            layoutData.Add("ProductsByCategory", SimpleSubReportExampleData.ProductsByCategory)

            ' Layout the document and set it's properties
            Dim document As Document = layoutReport.Layout(layoutData)
            document.Author = "DynamicPDF ReportWriter"
            document.Title = "Simple SubReport Example"

            ' Outputs the document to a file
            document.Draw(outputPdfPath)
        End Sub
    End Class
End Namespace

File: coresuite-dotnet-framework-console-cs/reportwriter/SimpleSubReport.cs
using ceTe.DynamicPDF;
using ceTe.DynamicPDF.LayoutEngine;

namespace coresuite_dotnet_framework_console_cs.ReportWriter
{
    public class SimpleSubReport
    {
        public static void Run(string outputPdfPath)
        {
            // Create the document's layout from a DLEX template
            DocumentLayout layoutReport = new DocumentLayout(Util.GetResourcePath("DLEX/SimpleSubReport.dlex"));

            NameValueLayoutData layoutData = new NameValueLayoutData();
            layoutData.Add("ProductsByCategory", SimpleSubReportExampleData.ProductsByCategory);

            // Layout the document and set it's properties
            Document document = layoutReport.Layout(layoutData);
            document.Author = "DynamicPDF ReportWriter";
            document.Title = "Simple SubReport Example";

            // Outputs the document to a file
            document.Draw(outputPdfPath);
        }
    }
}